src/Button/Button.component.tsx

```diff
@@ -3,113 +3,580 @@ import * as React from 'react';
+    ${({ forceActive, disabled, revealed, buttonStyles }: StyledButtonProps) =>
+        !disabled &&
+        !revealed &&
+        buttonStyles.active &&
+        css`
+            &:active {
+                ${buttonStyles.active}
+            }
+            ${forceActive && buttonStyles.active}
+        `}
+
+    /* Focus styles */
+    ${({ forceFocus, disabled, revealed, buttonStyles }: StyledButtonProps) =>
+        !disabled &&
+        !revealed &&
+        buttonStyles.focus &&
+        css`
+            &:focus {
+                ${buttonStyles.focus}
+            }
+            ${forceFocus && buttonStyles.focus}
+        `}
+
+    /* Outline */
+    &:focus {
+        ${OutlineStyles}
+    }
+    ${({ forceFocus, ...props }: StyledButtonProps) =>
+        forceFocus && OutlineStyles(props)}
+
+    /* Revealed State */
+    ${({ variant, revealed }: StyledButtonProps) =>
+        variant === 'primary' &&
+        revealed &&
+        css`
+            background-color: ${colors.silver.base};
+            border: solid thin ${colors.silver.base};
+            color: ${colors.charcoal.light};
+            font-weight: bold;
+        `}
+`;
+
+interface StyledHitboxProps {
+    buttonHeight: number;
+    buttonWidth?: number;
+}
+
+const StyledHitbox = styled.div<StyledHitboxProps>`
+    position: absolute;
+
+    // overlap border
+    top: -1px;
+    left: -1px;
+    right: -1px;
+    bottom: -1px;
+
+    // expand using margin to get to 3 rem tall and wide
+    margin: ${({ buttonHeight, buttonWidth }: StyledHitboxProps) =>
+        `-${(3 - buttonHeight) / 2}rem ${
+            buttonWidth && buttonWidth < 3
+                ? `-${(3 - buttonWidth) / 2}rem`
+                : '0'
+        }`};
+`;
+
+interface StyledFlipProps {
+    flipColor: string;
+}
+
+const StyledFlip = styled.div<StyledFlipProps>`
+    border-top-right-radius: ${sizes.border.radius.base};
+    border-bottom-left-radius: ${sizes.border.radius.base};
+
+    position: absolute;
+    top: -1px;
+    right: -1px;
+    width: 1.25rem;
+    height: 1.25rem;
+    content: '';
+
+    transition: ${transitionSpeed} ease opacity;
+    background: linear-gradient(
+        45deg,
+        ${({ flipColor }) => flipColor},
+        ${({ flipColor }) => flipColor} 50%,
+        ${colors.silver.base} 0
+    );
 `;
 
+const Flip = ({ colorTheme }: { colorTheme: Theme }) => (
+    <React.Fragment>
+        <StyledFlip flipColor={colorTheme.base} />
+        <StyledFlip className="flip-base" flipColor={colorTheme.light} />
+    </React.Fragment>
+);
+
 export const Button = ({
     className,
+    flip = false,
+    variant = 'primary',
+    size = 'large',
+    disabled,
+    revealed,
+    colorTheme,
+    reverse,
+    circular,
+    children,
+    icon: Icon,
     ...props
-}: ButtonProps): React.ReactElement<any> => (
-    <StyledButton
-        className={classNames('anchor-button', className)}
-        {...props}
-    />
-);
+}: ButtonProps): React.ReactElement<any> => {
```

Comment:
`}: ButtonProps): React.ReactElement<any> => {`
can be
`}: ButtonProps): React.ReactElement<ButtonProps> => {`